### ISSUE TITLE
persistence: TAIL-ing a persistent source immediately after restart reports no data, timestamp = 0

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#9217]
### What version of Materialize are you using?

From the shell:

```
25cce1b00e578175231935c500fae59b553e2acb
```

### What was the issue?

If a persisted source is TAIL-ed after restart, it will report no data and the timestamp column will have 0 in it. Both of those things are transient and go away, but should not have been reported in first place.

Instead, the source should return an error until it has been fully hydrated. This is what happens in our Postgres direct replication -- the source does not return any data until the initial snapshot has been replicated and consumed.

### Is the issue reproducible? If so, please provide reproduction instructions.

1. Pull the `test-for-gh9217` branch from `git@github.com:philip-stoev/materialize.git`
2. Run the test:
```
cd test/persistence
./mzcompose down -v
./mzcompose run kafka-sources
```

3. It will fail as follows:
```
kafka-sources/aaaa-linearizability-after.td:25:1: error: non-matching rows: expected:
[["<TIMESTAMP>", "1", "1000000"]]
got:
[["0", "1", "0"]]
Diff:
extra row: ["0", "1", "0"]
row missing: ["<TIMESTAMP>", "1", "1000000"]

     |
  24 | 
  25 | > FETCH 1 c;
     | ^
```

In other words, the test:
- expects FETCH will return a 13-digit timestamp and gets 0 instead
- expects that the source will contain 1M records, instead the materialized COUNT(*) view on top of the source reports 0 rows.

With kafka source persistence disabled:
- the timestamp is never 0, it is always a valid integer
- the count is 0, as expected since the source is yet to be rehydrated.

**Comment by philip-stoev** [comment]
Some thoughts by @mjibson about timestamp = 0 in another context : #6295

**Comment by aljoscha** [comment]
Closing this because current persist/storage architecture has moved quite a bit beyond this.

@philip-stoev Please yell at me if you'd like to keep this open.

fyi @danhhz
